python/pyarrow/array.pxi

```diff
@@ -3573,17 +3595,19 @@ class FixedShapeTensorArray(ExtensionArray):
           ]
         ]
         """
-        if not obj.flags["C_CONTIGUOUS"]:
-            raise ValueError('The data in the numpy array need to be in a single, '
-                             'C-style contiguous segment.')
+
+        permutation = (-np.array(obj.strides)).argsort()
```

Comment:
Neat, though I think we want a stable sort in case there are duplicates in `strides`:
```suggestion
        permutation = (-np.array(obj.strides)).argsort(kind='stable')
```

Reply:
Oh, good point!